### PR TITLE
Adding add-ons to the project

### PR DESCRIPTION
**Pull request by DijonnaWatson**
updating the inside and outside gradle.build and the gradle.yml. Trying out the new function to auto run the servers.

Prompt: start up servers from within gradle, and use that to create an automated end-to-end testing of the grpc logic that runs along with the rest of the tests in github for every pull request

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,53 +1,102 @@
...
+                commandLine './gradlew', 'test'
+            }
+        } finally {
+            // Stop both servers
+            dataStoreProcess.destroy()
+            factorServerProcess.destroy()
+            println "Both servers stopped after testing."
+        }
+    }
 }
```
Comment (ViceRaven) [inline build.gradle:102]: Ok its good we got the build.gradle to work. I didn't even realize we had a misplaced bracket but I'm glad we caught it.

**Review by ViceRaven — APPROVED** [review]

File: build.gradle [hunk 2]
```diff
@@ -1,53 +1,102 @@
...
+
+        println "Both servers started for testing..."
+
+        try {
+            // Run the tests
+            exec {
+                commandLine './gradlew', 'test'
+            }
+        } finally {
+            // Stop both servers
```
Comment (jeremy23newman) [inline build.gradle:96]: good job adding a command to stop the servers from continuing to run after the tests are done

**Review by jeremy23newman — APPROVED** [review]

**Comment by jeremy23newman** [comment]
we used pair programming on Dijonna's commit in order to do the add on feature in one commit

**Comment by ViceRaven** [comment]
> we used pair programming on Dijonna's commit in order to do the add on feature in one commit

To expand on this we all participated in the add-on feature during lab time. Even though this is dijonna's PR, Jeremy and I still collaborated with her in person to develop the add-on feature.

**Pull request merged**